todo-app/src/todo/service.ts

```diff
@@ -0,0 +1,87 @@
+import {
+    BadRequestException,
+    Injectable,
+    NotFoundException,
+  } from '@nestjs/common';
+  import { InjectRepository } from '@nestjs/typeorm';
+  import { Todo } from './entity/todo.entity';
+  import { Repository } from 'typeorm';
+  import { CreateTodoRequestDto } from './dto/request.dtos';
+  import { TodoResponseDto } from './dto/response.dtos';
+  import { startOfDay } from 'date-fns';
+  
+  @Injectable()
+  export class TodoService {
+    constructor(
+      @InjectRepository(Todo)
+      private todoRepository: Repository<Todo>,
+    ) {}
+  
+    private isValidTodoDate(todoDate: Date): Boolean {
+      const currentDate = new Date();
+      const inputTodoDate = new Date(todoDate);
+  
+      return startOfDay(currentDate) < startOfDay(inputTodoDate);
```

Comment:
- avoid using methods from these libraries directly in your code
- instead create a util and use this library there
- that way, if you have to change the underlying library, you would only need to change the util and the rest of your code would be perfectly intact